crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -455,6 +466,21 @@ impl Timeline {
         Some(found.clone())
     }
 
+    /// Edit an event given its [`TimelineEventItemId`] and some new content.
+    ///
+    /// See [`Self::edit`] for more information.
+    pub async fn edit_by_id(
```

Comment:
Can we extract this new method in a separate PR?

Reply:
Sure, I just added it because I wanted to move the logic down a layer where it could be tested, as I think @bnjbvr suggested in a chat room.